docs/topics/running/ambassador-with-aws.md

```diff
@@ -1,232 +1,364 @@
+   ```yaml
+   apiVersion: v1
+   kind: Service
+   metadata:
+     name: ambassador
+     namespace: ambassador
+     annotations:
+       service.beta.kubernetes.io/aws-load-balancer-ssl-cert: {{ACM_CERT_ARN}}
+       service.beta.kubernetes.io/aws-load-balancer-ssl-ports: "443"
+       service.beta.kubernetes.io/aws-load-balancer-backend-protocol: "http"
+   spec:
+     type: LoadBalancer
+     ports:
+     - name: HTTP
+       port: 80
+       targetPort: 8080
+     - name: HTTPS
+       port: 443
+       targetPort: 8080
+     selector:
+       service: ambassador
+   ```
+
+   Note that the `spec.ports` has been changed so both the HTTP and HTTPS ports forward to the cleartext port 8080 on Ambassador.
+
+* **Host:**
+   
+   The `Host` configures how Ambassador handles encrypted and cleartext traffic. The following `Host` configuration will tell Ambassador to `Redirect` cleartext traffic that comes in from the load balancer:
+
+   ```yaml
+   apiVersion: getambassador.io/v2
+   kind: Host
+   metadata:
+     name: ambassador
+   spec:
+     hostname: "*"
+     selector:
+       matchLabels:
+         hostname: wildcard
+     acmeProvider:
+       authority: none
+     requestPolicy:
+       insecure:
+         action: Redirect
+   ```
+
+* **Module:**
+
+   Since a L7 load balancer will be able to append to `X-Forwarded` headers, we need to configure Ambassador to trust the value of these headers. The following `Module` will configure Ambassador to trust a single L7 proxy in front of Ambassador:
+
+   ```yaml
+   apiVersion: getambassador.io/v2
+   kind: Module
+   metadata:
+     name: ambassador
+     namespace: ambassador
+   spec:
+     config:
+       xff_num_trusted_hops: 1
+       use_remote_address: false
+   ```
+
+**Note:**
+
+Ambassador uses the value of `X-Forwarded-Proto` to know if the request originated as encrypted or cleartext. Unlike L4 load balancers, L7 load balancers will set this header so HTTP -> HTTPS redirection is possible when terminating TLS at a L7 load balancer.
+
+## Preserving the Client IP Address
+
+Many applications will want to know the IP address of the connecting client. In Kubernetes, this IP address is often obscured by the IP address of the `Node` that is forwarding the request to Ambassador so extra configuration must be done if you need to preserve the client IP address.
+
+In AWS, there are two options for preserving the client IP address.
+
+1. Use a L7 Load Balancer that sets `X-Forwarded-For`
+
+   A L7 load balancer will populate the `X-Forwarded-For` header with the IP address of the downstream connecting client. If your clients are connecting directly to the load balancer, this will be the IP address of your client.
+
+   When using L7 load balancers, you must configure Ambassador to trust the value of `X-Forwarded-For`  and not append it's own IP address to it by setting `xff_num_trusted_hops` and `use_remote_address: false` in the [Ambassador `Module`](../ambassador):
```

Comment:
extra space before "   and not"?

Also, "it's own" -> "its own".

Reply:
Thanks!